sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -222,6 +225,14 @@ private[parquet] class ParquetFilters(pushDownDate: Boolean, pushDownStartWith:
     // See SPARK-20364.
     def canMakeFilterOn(name: String): Boolean = nameToType.contains(name) && !name.contains(".")
 
+    // All DataTypes that support `makeEq` can provide better performance.
+    def shouldConvertInPredicate(name: String): Boolean = nameToType(name) match {
```

Comment:
@HyukjinKwon  How about remove this?
`Timestamp` type and `Decimal` type will be support soon.

Reply:
Let us keep it.

Reply:
It depends on which PR will be merged first. The corresponding PRs should update this.

Reply:
Also need to update the benchmark suite.